### DISCUSSION TITLE
The table `table_dpid_seq_1` does not exist in the current database.

### DISCUSSION THREAD

**Discussion by guiqis** [prisma/prisma#25524] [Q&A]
### Question

I have a table in Postgres with columns id, account_id, display_id. account_id + display_id are an index. I've created a trigger and a function to populate display_id automatically based on last display_id for specific account_id. I also created the sequence to do it `CREATE SEQUENCE leads_dpid_seq_2 START WITH 1;`.

In pgadmin every works fine! When I insert a record with only account id, it populates the display_id correctly. But in prisma (query below) it returns `The table `table_dpid_seq_1` does not exist in the current database.`. Does anyone knows what I can do to solve this?

```
await prisma.table.create({ 
  data: {
     account_id: accountId
  }
})
```

**Comment by ludralph** [comment]
Hi @guiqis 👋 

The issue is because Prisma cannot access or recognize the sequence `leads_dpid_seq_2`. One way to resolve this is to use [customize migrations](https://www.prisma.io/docs/orm/prisma-migrate/workflows/customizing-migrations) to set up the sequence and triggers. To achieve this, you will:

1. Create the initial schema
```prisma
model Post {
  id          Int      @id @default(autoincrement())
  account_id  Int
  display_id  Int     
  @@index([account_id, display_id])
}
```
2. Run the command `npx prisma migrate dev --create-only` to allow for customizable migrations. Give the migration a name, for example, `init`, then press enter to create the migration file. Open the migration file and add the SQL code for the Sequence and the Trigger. The migration file should look like the below
```sql
-- CreateTable
CREATE TABLE "Post" (
    "id" SERIAL NOT NULL,
    "account_id" INTEGER NOT NULL,
    "display_id" INTEGER NOT NULL,

    CONSTRAINT "Post_pkey" PRIMARY KEY ("id")
);

-- CreateIndex
CREATE INDEX "Post_account_id_display_id_idx" ON "Post"("account_id", "display_id");

CREATE SEQUENCE leads_dpid_seq_2 START WITH 1;

CREATE OR REPLACE FUNCTION populate_display_id()
RETURNS TRIGGER AS $$
DECLARE
    max_display_id INT;
BEGIN
    -- Check if display_id is not provided
    IF NEW.display_id IS NULL THEN
        -- Get the maximum display_id for the given account_id
        SELECT COALESCE(MAX(display_id), 0) INTO max_display_id
        FROM "Post"
        WHERE account_id = NEW.account_id;

        -- If there is a max_display_id, increment it by 1
        -- Otherwise, get the next value from the sequence
        IF max_display_id > 0 THEN
            NEW.display_id := max_display_id + 1;
        ELSE
            NEW.display_id := nextval('leads_dpid_seq_2');
        END IF;
    END IF;

    RETURN NEW;
END;
$$ LANGUAGE plpgsql;

CREATE TRIGGER set_display_id
BEFORE INSERT ON "Post"
FOR EACH ROW
EXECUTE FUNCTION populate_display_id();
```
3. In the `schema.prisma` file, update the display_id to use `@default(dbgenerated("nextval('leads_dpid_seq_2')"))`. The schema file should now look like:
```prisma
model Post {
  id          Int      @id @default(autoincrement())
  account_id  Int
  display_id  Int     @default(dbgenerated("nextval('leads_dpid_seq_2')"))

  @@index([account_id, display_id])
}
```
4. You can now run the command `npx prisma migrate dev`. From here, you can go to the script, and execute the code 

```js
const result =  await prisma.post.create({ 
    data: {
       account_id: 1,
    }
  })
```
This should now correctly work as expected. sample output is shown below:
<img width="706" alt="image" src="https://github.com/user-attachments/assets/1f935930-0f3f-47d2-8e86-7c023bb9a8da">

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
